src/transformers/tokenization_utils_base.py

```diff
@@ -1730,27 +1730,39 @@ def from_pretrained(cls, pretrained_model_name_or_path: Union[str, os.PathLike],
 
         # Get files from url, cache, or disk depending on the case
         resolved_vocab_files = {}
+        unresolved_files = []
         for file_id, file_path in vocab_files.items():
             if file_path is None:
                 resolved_vocab_files[file_id] = None
             else:
                 try:
-                    resolved_vocab_files[file_id] = cached_path(
-                        file_path,
-                        cache_dir=cache_dir,
-                        force_download=force_download,
-                        proxies=proxies,
-                        resume_download=resume_download,
-                        local_files_only=local_files_only,
-                        use_auth_token=use_auth_token,
-                    )
+                    try:
+                        resolved_vocab_files[file_id] = cached_path(
+                            file_path,
+                            cache_dir=cache_dir,
+                            force_download=force_download,
+                            proxies=proxies,
+                            resume_download=resume_download,
+                            local_files_only=local_files_only,
+                            use_auth_token=use_auth_token,
+                        )
+                    except FileNotFoundError:
+                        if local_files_only:
+                            unresolved_files.append(file_id)
```

Comment:
Probably not necessary, but would a:

```
else:
    raise FineNotFoundError(...)
```

be helpful here?

Reply:
While it doesn't change anything right now as the only way a `FileNotFoundError` would be excepted here is it `local_files_only` is `True`
https://github.com/huggingface/transformers/blob/10518fe494f2ad683d72e7c174eeac9c309ce387/src/transformers/file_utils.py#L1227-L1232

It would indeed be more future proof if we add `FileNotFoundError`s down the road :) Updating.

Reply:
See 863e553